### ISSUE TITLE
ChA can select location for a chapter

### ISSUE THREAD

**Issue by dboyer** [Iridescent-CM/technovation-app#4684]
The chapter location tab of the ChA dashboard currently displays the country, region, and city served by the chapter. The country is pre-set by a Technovation admin (#4683). 

ChAs should be able to select the region and city. The region list is populated based on the country, and the city list is populated based on the region.

<img width="798" alt="Screen Shot 2024-04-30 at 11 27 42 AM" src="https://github.com/Iridescent-CM/technovation-app/assets/516764/866e3e1b-e6a5-4f16-b344-83930e005784">

**Comment by dboyer** [comment]
NOTE: We have decided that the ChA can set the whole location for the chapter, including the country. We should add the same location editing form that we use in other locations 

This is the form for admins to edit location. The same partial can be added to the "chapter location" screen on ChA's dashboards. 
<img width="1178" alt="Screen Shot 2024-10-24 at 11 38 29 AM" src="https://github.com/user-attachments/assets/542a68d6-6a94-4236-ad63-9ac0be1a1611">

When a ChA edits their location, we should show a pop-up that requires them to confirm the location change. Include this text:

The chapter location determines which users see your chapter as a suggested chapter when they register. Students and mentors see a list of chapters in their local area or their country. By editing your location, you understand that it may change which users can join your chapter when they register. Would you like to save this location?

Buttons: Save Location and Cancel

**Comment by viviancan** [comment]
<img width="800" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOE9GQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--ab1bbc68829f5105a36212d6486e0624b6faf8db/CleanShot%202024-10-30%20at%2009.06.41.gif" alt="CleanShot 2024-10-30 at 09.06.41.gif" />

**Comment by viviancan** [comment]
@dboyer This is on QA

**Comment by dboyer** [comment]
Tested on QA and confirmed:

- When I log in as a ChA and go to my Chapter Location tab, I now see an option to set the chapter location
- When I click to set the location, I get a warning message that changing the location will impact who sees my chapter as an option
- I see the location form and it works as expected when I enter a location
- I checked the admin page for the chapter and it now listed the new location I entered in the chapter location field

**Comment by dboyer** [comment]
Tested updating my location:

- As a mentor, I was able to update my location on my profile. As an admin, I saw the new location for that mentor on the mentor's admin participant page
- As a ChA, I changed my personal location and saw it had updated. I confirmed that it did not impact the location assigned to the chapter.
- As a student, I changed my location and saw it had updated.
- As a student on a team, I was able to update the team's location
